src/SFML/Network/IpAddress.cpp

```diff
@@ -210,7 +225,16 @@ bool operator !=(const IpAddress& left, const IpAddress& right)
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    if (left.m_valid < right.m_valid)
+        return true;
+
+    if (right.m_valid < left.m_valid)
+        return false;
+
+    if (left.m_address < right.m_address)
+        return true;
+
+    return false;
 }
```

Comment:
The operations are equivalent.

Definition of `std::pair<T1, T2>::operator<` from the standard:
> Returns: `x.first < y.first || (!(y.first < x.first) && x.second < y.second)`.